### DISCUSSION TITLE
Change type of variable without redefining

### DISCUSSION THREAD

**Discussion by Hedwig7s** [teal-language/tl#794] [Q&A] [answered]
Because of the fact that the current version of teal doesn't have type intersection I have to make new types, but of course the compiler doesn't know this so I have to redefine it
Is there a better way to do this?
Example:
```
local function example(ex:Foo)
    local ex = ex as Bar
end

**Comment by Hedwig7s** [comment]
Note: I forgot to mention that example is used somewhere where the type is already defined as Foo (say in a record) and as such can't have the input changed to Bar

**Comment by hishamhm** [comment, marked as answer]
No, not in `master`. But do give Teal `next` a spin and give interfaces a try!

If you're using LuaRocks, you can install it with `luarocks install tl --dev --branch next`. If you're installing it manually, you can grab `tl.tl` from the `next` branch at GitHub.

**Reply by Hedwig7s** [reply]
I'd love to, but I'm dependent on the build system which doesn't support next

**Reply by hishamhm** [reply]
Do you mean `cyan` or `tl build`? It's my understanding that `next` currently breaks `cyan`, but I'll get that fixed (on Teal's end)  ASAP.

**Reply by Hedwig7s** [reply]
Either way cyan doesn't support it in general and tl build is missing outright

**Reply by hishamhm** [reply]
My question was whether the build system you're currently using is `cyan` or `tl build`. `tl build` is deprecated and if you're using that you should switch to cyan; I will make sure that Teal next will support `cyan` prior to release.
